packages/ui-components/src/components/SearchInput/SearchInput.component.tsx

```diff
@@ -0,0 +1,226 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Juno contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useEffect, ChangeEvent, KeyboardEvent, MouseEvent } from "react"
+
+import { Icon } from "../Icon"
+import { Stack } from "../Stack"
+
+import "./searchinput.scss"
+
+export interface SearchInputProps {
+  /**
+   * Specifies the name attribute for the input element.
+   */
+  name?: string
+  /**
+   * Determines the visual styling variant of the SearchInput component.
+   * - "default": Standard search input styling.
+   * - "hero": A larger search input intended for standalone use on a dedicated search page, akin to the initial Google search page.
+   * - "rounded": A search input with rounded edges.
+   */
+  variant?: "rounded" | "hero" | "default"
+  /**
+   * Disables the search input when set to true.
+   */
+  disabled?: boolean
+  /**
+   * Custom placeholder text displayed in the search input.
+   */
+  placeholder?: string
+  /**
+   * Initial value for the search input. Note that this value will not be updated if changed by the user.
+   */
+  value?: string
+  /**
+   * Controls the autocomplete attribute of the input element.
+   * Pass a valid autocomplete value.
+   * We do not enforce validity.
+   * */
+  autoComplete?: string
+  /**
+   * Determines whether to show the 'Clear' button.
+   */
+  clear?: boolean
+  /**
+   * Pass an optional CSS class to apply to the search input.
+   */
+  className?: string
+  /**
+   * Callback function invoked when a search is triggered, either by pressing the 'Enter' key or by clicking the search icon.
+   * */
+  onSearch?: (_value: string) => void
+  /**
+   * Click handler for the search icon.
+   */
+  onClick?: (_event: MouseEvent<HTMLButtonElement | HTMLAnchorElement>) => void
+  /**
+   * Change handler for the search input.
+   */
+  onChange?: (_event: ChangeEvent<HTMLInputElement>) => void
+  /**
+   * KeyPress handler for the search input. By default, triggers the onSearch function when the 'Enter' key is pressed.
+   */
+  onKeyPress?: (_event: KeyboardEvent<HTMLInputElement>) => void
```

Comment:
FYI that `onKeyPress` is deprecated https://github.com/cloudoperators/juno/issues/247#issuecomment-2418881656